### ISSUE TITLE
Broken in Julia v0.7

### ISSUE THREAD

**Issue by epatters** [kmsquire/Match.jl#47]
Looks like Match.jl suffered some breakage, probably fairly minor, in Julia v0.7:

```
(v0.7) pkg> test Match
   Testing Match
    Status `/var/folders/s4/c_2398351wz92ls6b777my080000gn/T/tmpO67aX6/Manifest.toml`
  [7eb4fadd] Match v0.4.0
WARNING: Base.Test is deprecated, run `using Test` instead
  likely near /Users/epatters/.julia/packages/Match/hykdg/test/matchtests.jl:2
ERROR: LoadError: LoadError: syntax: invalid let syntax
Stacktrace:
 [1] include at ./boot.jl:317 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1038
 [3] include(::Module, ::String) at ./sysimg.jl:29
 [4] include(::String) at ./client.jl:398
 [5] top-level scope at none:0
 [6] include at ./boot.jl:317 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1038
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] include(::String) at ./client.jl:398
 [10] top-level scope at none:0
in expression starting at /Users/epatters/.julia/packages/Match/hykdg/test/matchtests.jl:41
in expression starting at /Users/epatters/.julia/packages/Match/hykdg/test/runtests.jl:3
ERROR: Package Match errored during testing
```

The issue above is due to a change in the AST representation of `let` blocks (JuliaLang/julia#21774).

**Comment by rfrancis** [comment]
There's also use of Base.ismatch, which was deprecated out of existence, at least in matchutils.jl.  And so:

```
ERROR: LoadError: LoadError: UndefVarError: ismatch not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/russ/.julia/packages/Match/hykdg/src/Match.jl:2
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] top-level scope at none:2
 [12] eval at ./boot.jl:319 [inlined]
 [13] eval(::Expr) at ./client.jl:389
 [14] top-level scope at ./none:3
in expression starting at /home/russ/.julia/packages/Match/hykdg/src/matchutils.jl:8
in expression starting at /home/russ/.julia/packages/Match/hykdg/src/Match.jl:8
ERROR: LoadError: Failed to precompile Match [7eb4fadd-790c-5f42-8a69-bfa0b872bfbf] to /home/russ/.julia/compiled/v1.0/Match/XxUFO.ji.

```

**Comment by epatters** [comment]
Resolved by #49. Thanks @kmsquire!

**Comment by ralphmrm1** [comment]
This from Julia 1.1.0 (code from [https://en.wikibooks.org/wiki/Introducing_Julia/DataFrames](url) )

  `ptable[[ismatch(r"Copper", elementname) for elementname in ptable[:Name]], :][:Liquid] = 2567 - 1083`

UndefVarError: ismatch not defined
Stacktrace:
 [1] (::getfield(Main, Symbol("##35#36")))(::String) at .\none:0
 [2] iterate at .\generator.jl:47 [inlined]
 [3] collect(::Base.Generator{Array{String,1},getfield(Main, Symbol("##35#36"))}) at .\array.jl:606
 [4] top-level scope at In[77]:1
